PusherServer/RequestResult.cs

```diff
@@ -10,6 +11,11 @@ public abstract class RequestResult: IRequestResult
 
         public RequestResult(IRestResponse response)
         {
+            if (response == null)
+            {
+                throw new ArgumentNullException("response");
+            }
+
```

Comment:
I don't remember what I was doing when I added this. But, purely based on looking at it I would say you get a clearer error from this if `null` is passed. It will tell you that the `response` parameter is null. If that check weren't there then accessing `response.Content` would result in an exception that says `Object reference not set to an instance of an object.` which isn't as clear.